lib/distribute.c

```diff
@@ -419,8 +472,8 @@ int config_show_distribute(struct vty *vty)
 	else
 		vty_out(vty, " not set\n");
 
-	for (i = 0; i < disthash->size; i++)
-		for (mp = disthash->index[i]; mp; mp = mp->next) {
+	for (i = 0; i < dist_ctx.disthash->size; i++)
+		for (mp = dist_ctx.disthash->index[i]; mp; mp = mp->next) {
```

Comment:
use the `distribute_ctx` given as parameter and not the global `dist_ctx`, which should be removed